controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -594,17 +618,21 @@ func (r *KubeadmControlPlaneReconciler) reconcileExternalReference(ctx context.C
 		return err
 	}
 
+	if util.IsPaused(cluster, obj) {
+		return errors.Wrapf(&capierrors.RequeueAfterError{RequeueAfter: 30 * time.Second},
```

Comment:
Why requeue after? Aren't we watching the external kind? Or is this the only current way to short-circuit the reconcile so it doesn't continue doing anything else?

Reply:
We aren't currently watching the external resource. Since there are no real actions that we would need to take if the Infrastructure template changes (we assume it to be immutable), it would seem odd that we would add a watch for no other reason than testing if a previously paused external referenced resource is no longer unpaused.

Reply:
Can we instead say we don't look at pause on templates?

Reply:
Yeah I would agree with this, it would be in line with what the MS/MD does today. Templates shouldn't be reconciled anyways, so having the paused annotation on it shouldn't do anything.

Reply:
I'm good with that as well.